chain/chain/src/chain.rs

```diff
@@ -2430,6 +2430,13 @@ impl<'a> ChainUpdate<'a> {
         let head = self.chain_store_update.head()?;
         let is_next = block.header.prev_hash == head.last_block_hash;
 
+        // Check that we know the epoch of the block before we try to get the header
+        // (so that a block from unknown epoch doesn't get marked as an orphan)
+        if let Err(_) = self.runtime_adapter.get_epoch_inflation(&block.header.inner_lite.epoch_id)
```

Comment:
So the inflation method is the only one that touches the epoch today.
I can just add another method into runtime adapter that checks if an epoch is known, that would be less clutchy, but more methods in the runtime adapter, which is already pretty big.